### DISCUSSION TITLE
User rights not saying if they are enabled or disabled

### DISCUSSION THREAD

**Discussion by va3ets200** [BearWare/TeamTalk5#2373] [Q&A]
I'm running the latest 5.17 beta, 5.17.0.5144, and in the user rights section of the new user dialog, I assume this is the same when editing an existing account, with voiceOver on my Mac here, it is not at all spoken whether a user right  is enabled or disabled, so as of now, the VoiceOver user isn't going to know.  As well, how would I go about editing an existing account?  all I can seem to fine, is the list of users, and a new user button.  If the option to edit an account, is in a context menu, access to context menus, which we currently do not have on the Mac with VoiceOver, is extremely crucial.

**Comment by bear101** [comment]
@CoBC if I enable Qt::AccessibleTextRole in UserRightsModel then VoiceOver announce the text message:
```
#if QT_VERSION < QT_VERSION_CHECK(6,0,0)
    case Qt::AccessibleTextRole :
        switch (m_userrights[index.row()])
        {
        case USERRIGHT_LOCKED_NICKNAME :
        case USERRIGHT_LOCKED_STATUS :
            return QString("%1: %2").arg(data(index, Qt::DisplayRole).toString()).arg((m_activeUserRights & m_userrights[index.row()])? tr("Disabled") : tr("Enabled"));
        default :
            return QString("%1: %2").arg(data(index, Qt::DisplayRole).toString()).arg((m_activeUserRights & m_userrights[index.row()])? tr("Enabled") : tr("Disabled"));
        }
#endif
```

**Comment by CoBC** [comment]
Oh, maybe checkstateRole is only accessible on Windows. In this case, we 
should change the check to OS and not only Qt version

**Reply by bear101** [reply]
Yes, I tried TTSEventsModel in TeamTalk v5.16 on macOS and it also doesn't announce checked/unchecked.

**Reply by bear101** [reply]
so I suggest

`#if QT_VERSION < QT_VERSION_CHECK(6,0,0) || defined(Q_OS_MAC)`

**Comment by CoBC** [comment]
OK, I'll change the check for all models

**Comment by CoBC** [comment]
Yes, it's also what I thought :)

**Comment by CoBC** [comment]
Done in 1d6eb9d

Unfortunately I don't see any easy way to fix the context menu issue.

A bug is open on the Qt bug tracker, but without any activity recently :(

**Comment by bear101** [comment]
@va3ets200 for context menu on macOS I use "Alternative pointer actions". This brings up the context menu (same as right click).

https://iboysoft.com/wiki/context-menu-mac.html

**Reply by bear101** [reply]
I've set F12 as right-click

**Comment by CoBC** [comment]
FYI to follow progress:
https://bugreports.qt.io/browse/QTBUG-68465
I've added a comment to ask for any news on the topic, and also contacted 2 Qt devs to see if they can helps

**Comment by va3ets200** [comment]
Oh wow, that’s cool.  Wasn’t aware of that accessibility setting in system settings.  Thanks for that article.  Hopefully that’ll help in the mean time, while the QT bug, is still a thing

**Comment by va3ets200** [comment]
Actually in looking at that report, the keystroke in VO to access context menus, would be VO-Shift-M, not VO-Shift-down arrow.  Just thought I’d let you know about that.

**Comment by CoBC** [comment]
Hi,
FYI, the issue with context menu on MacOS has been fixed in Qt. The fix will probably be integrated in Qt6.8 beta 2

**Comment by va3ets200** [comment]
Ok, that’s good.  When can we expect this to be in a version of TeamTalk then?
